src/Constraint/Constraint.php

```diff
@@ -171,6 +171,73 @@ public function versionCompare($a, $b, $operator, $compareBranches = false)
         return version_compare($a, $b, $operator);
     }
 
+    public function compile($otherOperator) {
+        $otherOperator = self::$transOpStr[$otherOperator];
+
+        $isBranch = 'dev-' === substr($this->version, 0, 4);
```

Comment:
This could also expressed using `[]` string offsets in case blackfire thinks substr() is too hot

Reply:
done